include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,318 @@
+    {
+        return translate_single_fn_v(std::forward<irng_t>(irange));
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\brief A view performs single frame translation of nucleotide into aminoacid alphabet
+ * \tparam irng_t The type of the range being processed.
+ * \param irange The range being processed.
+ * \returns A range containing aminoacid sequence. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to translate nucleotide sequences into single forward aminoacid sequences.
+
+ * \par View properties
+ *
+ * |                     | `irng_t` (range input type)           | `rrng_t` (range return type)                              |
+ * |---------------------|---------------------------------------|-----------------------------------------------------------|
+ * | range               | seqan3::input_range_concept           | seqan3::view_concept + all range concepts met by `irng_t` |
+ * |                     | optional: seqan3::sized_range_concept |                                                           |
+ * | `range_reference_t` | seqan3::nucleotide_concept            |                                                           |
+ *
+ * * `irng_t` is the type of the range modified by this view (input).
+ * * `rrng_type` is the type of the range returned by this view.
+ * * for more details, see \ref view.
+ *
+ * \par Example
+ *
+ * Operating on a range of seqan3::dna5:
+ * ```cpp
+ * dna5_vector vec{"ACGTACGTACGTA"_dna5};
+ *
+ * // single frame translation
+ * auto v1 = vec | view::translate_single;                     // == [T,Y,V,R]
+ *
+ * // function syntax
+ * auto v2 = view::translate_single(vec);                      // == [T,Y,V,R]
+ *
+ * // combinability
+ * auto v3 = vec | view::complement | view::translate_single;  // == [C,M,H,A]
+ * ```
+ */
+seqan3::detail::translate_single_fn const translate_single;
+} // namespace seqan3::view
+
+
+namespace seqan3
+{
+enum translation_frames
+{
+    SINGLE_FRAME             = 1,
+    WITH_REVERSE_COMPLEMENT  = 2,
+    WITH_FRAME_SHIFTS        = 3,
+    SIX_FRAME                = 6
+};
+}
+
+namespace seqan3::detail
+{
+struct translate_frames_fn
+{
+    /*!\brief Single, reverse, forward-frame or six-frame translation from nucleotide to aminoacid.
+     * \tparam irng_t The type of the range being processed.
+     * \param irange The range being processed.
+     * \param tf Translation frames (SINGLE_FRAME, WITH_REVERSE_COMPLEMENT, WITH_FRAME_SHIFTS, SIX_FRAME).
+     */
+    template <typename irng_t>
+        requires forward_range_concept<irng_t> && nucleotide_concept<std::decay_t<ranges::range_reference_t<std::decay_t<irng_t>>>>
+    auto operator()(irng_t && irange, translation_frames const & tf) const
+    {
+        std::vector<ranges::any_random_access_view<aa27> > frames;
```

Comment:
we can leave it like this right now, but please add a `TODO: change to size-preserving view later`